plugins/outputs/prometheus_client/prometheus_client.go

```diff
@@ -123,7 +123,13 @@ func (p *PrometheusClient) Init() error {
 		ipRange = append(ipRange, ipNet)
 	}
 
-	authHandler := internal.BasicAuthHandler(p.BasicUsername, p.BasicPassword, "prometheus", onAuthError)
+	password, err := p.BasicPassword.Get()
+	if err != nil {
+		return err
+	}
+	defer config.ReleaseSecret(password)
+
+	authHandler := internal.BasicAuthHandler(p.BasicUsername, string(password), "prometheus", onAuthError)
```

Comment:
Can you please store the secret in the string and directly release the secret!? The reason is to free the locked memory as quick as possible as only a small portion is available usually...
```suggestion
	password, err := p.BasicPassword.Get()
	if err != nil {
		return err
	}
	passwd := string(password)
	config.ReleaseSecret(password)

	authHandler := internal.BasicAuthHandler(p.BasicUsername, passwd, "prometheus", onAuthError)
```

Reply:
Great catch! I edited the PR.